src/overlays/actors/ovl_En_Bigokuta/z_en_bigokuta.c

```diff
@@ -28,99 +43,640 @@ const ActorInit En_Bigokuta_InitVars = {
+                rotX = -0xC00;
+            } else if (this->timer > 0) {
+                rotX = 0x1800;
+            } else {
+                rotX = ((this->timer + 0x18) * 0.041666668f * -6144.0f);
+            }
+            rot->x -= rotX;
+        }
+    } else if (limbIndex == BIGOKUTA_LIMB_SNOUT_CENTER) {
+        EnBigokuta* this = THIS;
+
+        if (this->actionFunc == EnBigokuta_PlayDeathEffects) {
+            if (this->timer < 5) {
+                Matrix_Scale(1.0f, 1.0f, (this->timer * 0.2f * 0.25f) + 1.0f, MTXMODE_APPLY);
+            } else if (this->timer < 8) {
+                f32 time = (this->timer - 5) * (1 / 12.0f);
+
+                Matrix_Scale(1.0f + time, 1.0f + time, 1.25f - time, MTXMODE_APPLY);
+            } else {
+                Matrix_Scale(1.25f - ((this->timer - 8) * 0.125f), 1.25f - ((this->timer - 8) * 0.125f), 1.0f,
+                             MTXMODE_APPLY);
+            }
+        } else if (this->actionFunc == EnBigokuta_SuckInPlayer) {
+            f32 sin = sin_rad(this->timer * (M_PI / 3.0f)) * 0.5f;
+
+            Matrix_Scale(((this->timer * (2 / 90.0f)) * (0.5f + sin)) + 1.0f,
+                         ((this->timer * (2 / 90.0f)) * (0.5f - sin)) + 1.0f, 1.0f - ((this->timer * 0.3f) / 9.0f),
+                         MTXMODE_APPLY);
+        } else if (this->actionFunc == EnBigokuta_HoldPlayer && this->timer != 1) {
+            if (this->timer == 0) {
+                Matrix_Scale(0.9f, 0.9f, 1.15f, MTXMODE_APPLY);
+            } else if (this->timer > 0) {
+                f32 sin = sin_rad(this->timer * (M_PI / 3.0f)) * 0.5f;
+
+                Matrix_Scale(((0.5f + sin) * 0.2f) + 1.0f, ((0.5f - sin) * 0.2f) + 1.0f, 0.7f, MTXMODE_APPLY);
+            } else {
+                Matrix_Scale(1.0f - ((this->timer + 0x18) * 0.2f * 0.041666668f),
+                             1.0f - ((this->timer + 0x18) * 0.2f * 0.041666668f),
+                             ((this->timer + 0x18) * 0.3f * 0.041666668f) + 1.0f, MTXMODE_APPLY);
+            }
+        }
+    }
+    return false;
+}
+
+void EnBigokuta_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx,
+                             Gfx** gfx) {
+    static s8 D_80AC45BC[] = {
+        -1, -1, -1, 0, -1, 1, -1, 2, -1, 3, 8, 4, -1, 5, -1, -1, -1, -1, 6, 7,
+    };
+    static Vec3f D_80AC45D0[] = {
+        { 0.0f, 2000.0f, 1000.0f },    { 0.0f, 2000.0f, -2000.0f }, { 1700.0f, 700.0f, -600.0f },
+        { -1700.0f, 700.0f, -600.0f }, { 0.0f, 500.0f, -2500.0f },
+    };
+    EnBigokuta* this = THIS;
+    s32 i;
+    s8 limbPosIndex = D_80AC45BC[limbIndex];
+
+    if (limbPosIndex != -1) {
+        if (limbPosIndex < 6) {
+            Matrix_GetStateTranslationAndScaledX(800.0f, &this->limbPos[limbPosIndex]);
+        } else if (limbPosIndex < 8) {
+            Matrix_GetStateTranslation(&this->limbPos[limbPosIndex]);
+        } else {
+            for (i = 0; i < ARRAY_COUNT(this->limbPos) - 8; i++) {
```

Comment:
This loop goes up to 5, right? I think it would make a bit more sense to do a `ARRAY_COUNT(D_80AC45D0)` instead
```suggestion
            for (i = 0; i < ARRAY_COUNT(D_80AC45D0); i++) {
```